### PR TITLE
Improve Handling for TLS

### PR DESCRIPTION
**Pull request by sheldonhull**
## Type of Change
 - [x] Bug fix

### Purpose
AWS instances with restricted internet settings fail to process webrequest without basic parsing. A better practice for rest based requests is to use `Invoke-RestMethod`. 

### Approach
- Timeout parameter added for allowing override of AWS metadata
- New internal function cloned from `Invoke-TlsWebRequest` that is using `Invoke-RestMethod`
- Proper handling of failure to connect with `try{} catch{}` should ensure timeout exception handled properly and warned, bypassing attempt to connect to AWS metadata further.
- *new* added 3 properties related to CPU metrics to improve the results for Get-DbaComputerSystem in reviewing basics of OS performance specs

### Commands to test
1. Get-DbaComputerSystem
2. Invoke-TlsRestMethod *new internal helper*

# Note
1. Without AWS infrastructure can't verify this with pester test, and didn't see any `Invoke-TLS*` tests built. The new error handling worked, and confirmed on my own AWS instance that results pulled correctly, but you'll want to verify as well. 
2. Not sure of any other location to add detail to.

### CODE REVIEW

File: functions/Get-DbaComputerSystem.ps1 [hunk 1]
```diff
@@ -55,6 +58,7 @@ function Get-DbaComputerSystem {
         [DbaInstanceParameter[]]$ComputerName = $env:COMPUTERNAME,
         [PSCredential]$Credential,
         [switch]$IncludeAws,
+        [int]$TimeoutSec = 5,
```
Comment (wsmelton) [inline functions/Get-DbaComputerSystem.ps1]: Why does this need to be a parameter? As well, we had it set to 15 previously, if the parameter is to stay in place it needs to be set to 15 seconds.

Comment (wsmelton) [reply functions/Get-DbaComputerSystem.ps1]: Talking with Chrissy this needs to be removed and implemented as a module configuration. 

However from my point of view I see no reason for it. The site you are hitting is a local site in AWS servers only. I believe we can simply set it to a value and leave it.

**Comment by wsmelton** [comment]
When the timeout parameter is removed I'm good for this to be merged.

File: functions/Get-DbaComputerSystem.ps1 (hunk 1 above)
Comment (sheldonhull) [reply functions/Get-DbaComputerSystem.ps1]: Sounds good to me. thanks!

**Comment by wsmelton** [comment]
I have not had time but Azure finally added an REST API on Azure VMs for similar information if you happen to want to tackle that one 😁

**Comment by wsmelton** [comment]
Failure was unrelated on Appveyor

**Comment by niphlod** [comment]
BTW: this sparked a bug in https://github.com/sqlcollaborative/dbatools/issues/5336 . Remote instances do not have our own local commands, so invoke-command2 on anything other than localhost fails miserably.

I'll see what can be done in terms of "remoting" the two needed functions

**Pull request merged**